default.nix

```diff
@@ -9,16 +9,18 @@ let
 in
 rec {
 
-  hydra-cli = ((pkgs.callPackage ./Cargo.nix {
-    cratesIO = pkgs.callPackage ./crates-io.nix {};
-  }).hydra_cli {}).overrideDerivation(_: {
-    src = sources;
-    doCheck = true;
-    checkPhase = ''
-      echo "Checking formatting with 'rustfmt'"
-      find . -name "*.rs" | xargs ${rustfmt}/bin/rustfmt --check
-    '';
-  });
+  # XXX: we use buildPackageSingleStep instead of buildPackage, because the
+  # cargo version is too old to benefit from incremental builds.
+  hydra-cli = pkgs.naersk.buildPackageSingleStep (pkgs.lib.cleanSource ./.)
```

Comment:
Does this means we need to rebuild all Rust dependencies on each `sources` change (the same behavior than `buildRustPackage`)?
If true, would it be possible to update cargo and switch to an incremental build?
Because I'd prefer to keep the build as efficient as possible, even if build expressions are more complex.